File: reactjs/user-app/src/services/UserService.js
import axios from 'axios';

const USERS_REST_API_URL = 'http://localhost:8080/api';

class UserService{

    getUsers(){
        USERS_REST_API_URL = USERS_REST_API_URL + '/users';
        return axios.get(USERS_REST_API_URL);
    }
    getUserByEmailAndPassword(email,password){
        USERS_REST_API_URL = USERS_REST_API_URL + '/getUserByEmailAndPassword/'+email+'/'+password;
        return axios.get(USERS_REST_API_URL);
    }
    getUserById(Id){
        USERS_REST_API_URL = USERS_REST_API_URL + '/getUserById/'+Id;
        return axios.get(USERS_REST_API_URL);
    }
}

export default new UserService();

File: demo/src/main/java/com/user/service/UserService.java
package com.user.service;

import java.util.List;

import com.user.exception.UserNotFoundException;
import com.user.model.User;


public interface UserService {
	
	List<User> getUsers() ;

	User getUserById(Long userId) throws UserNotFoundException;

	List<User> getUserByEmailAndPassword(String email,String password) throws UserNotFoundException;
	
}
